obal/data/playbooks/nightly/nightly.yaml

```diff
@@ -13,21 +13,52 @@
       when: nightly_releaser is defined and '--arg jenkins_job=' in (build_package_tito_releaser_args | join(' '))
 
     - block:
-      - name: 'set nightly_macro_args'
-        set_fact:
-          nightly_macro_args:
-            - "--arg rpmbuild_options=\"--define 'nightly .{{ lookup('pipe', 'date +%Y%m%d%H%M%S') }}git{{ nightly_githash[:7] }}'\""
-
-      - name: 'append nightly_macro_args to build_package_tito_releaser_args'
-        set_fact:
-          build_package_tito_releaser_args: "{{ build_package_tito_releaser_args }} + {{ nightly_macro_args }}"
-
-      - name: copy source files
-        copy:
-          src: "{{ item }}"
-          dest: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
-        with_items: "{{ nightly_sourcefiles }}"
+        - name: 'set package_dir'
+          set_fact:
+            package_dir: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
+
+        - name: 'set global_nightly_macro'
+          set_fact:
+            global_nightly_macro: "%global nightly .{{ lookup('pipe', 'date +%Y%m%d%H%M%S') }}git{{ nightly_githash[:7] }}"
+
+        - name: 'copy source files'
+          copy:
+            src: "{{ item }}"
+            dest: "{{ package_dir }}"
+          with_items: "{{ nightly_sourcefiles }}"
+
+        - name: 'get nightly_specfile'
```

Comment:
might be out of scope for this PR, but this could use some refactoring too

```
% git grep -A2 -B2 "pattern:.*spec"
obal/data/playbooks/bump-release/bump-release.yaml-    - name: get spec file
obal/data/playbooks/bump-release/bump-release.yaml-      find:
obal/data/playbooks/bump-release/bump-release.yaml:        pattern: "*.spec"
obal/data/playbooks/bump-release/bump-release.yaml-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
obal/data/playbooks/bump-release/bump-release.yaml-      register: spec_file
--
obal/data/playbooks/changelog/changelog.yaml-    - name: get spec file
obal/data/playbooks/changelog/changelog.yaml-      find:
obal/data/playbooks/changelog/changelog.yaml:        pattern: "*.spec"
obal/data/playbooks/changelog/changelog.yaml-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
obal/data/playbooks/changelog/changelog.yaml-      register: spec_file
--
obal/data/playbooks/update/update.yaml-    - name: get spec file
obal/data/playbooks/update/update.yaml-      find:
obal/data/playbooks/update/update.yaml:        pattern: "*.spec"
obal/data/playbooks/update/update.yaml-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
obal/data/playbooks/update/update.yaml-      register: spec_file
--
obal/data/roles/diff_package/tasks/git_package_info.yml-- name: get spec file
obal/data/roles/diff_package/tasks/git_package_info.yml-  find:
obal/data/roles/diff_package/tasks/git_package_info.yml:    pattern: "*.spec"
obal/data/roles/diff_package/tasks/git_package_info.yml-    path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
obal/data/roles/diff_package/tasks/git_package_info.yml-  register: spec_file
--
obal/data/roles/fetch_sources/tasks/main.yml-    - name: get spec file
obal/data/roles/fetch_sources/tasks/main.yml-      find:
obal/data/roles/fetch_sources/tasks/main.yml:        pattern: "*.spec"
obal/data/roles/fetch_sources/tasks/main.yml-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
obal/data/roles/fetch_sources/tasks/main.yml-      register: spec_file
--
obal/data/roles/setup_sources/tasks/specfile.yml-- name: get spec file
obal/data/roles/setup_sources/tasks/specfile.yml-  find:
obal/data/roles/setup_sources/tasks/specfile.yml:    pattern: "*.spec"
obal/data/roles/setup_sources/tasks/specfile.yml-    path: "{{ package_dir }}"
obal/data/roles/setup_sources/tasks/specfile.yml-  register: setup_sources_spec_file

```

Reply:
We should also talk about enforcing the directory name matches the package name. Right now we don't care but IMHO we should prefer to follow the Fedora best practices and have them match. With that in mind I do think we should leave it to a separate PR to follow up on that.